mobile_packet_verifier/src/burner.rs

```diff
@@ -0,0 +1,266 @@
+        Ok(Self {
+            program_cache: BurnProgramCache::new(settings, &provider).await?,
+            provider,
+            valid_sessions,
+            keypair: keypair.to_bytes(),
+            cluster: settings.cluster.clone(),
+        })
+    }
+
+    pub async fn burn(&self, pool: &Pool<Postgres>) -> Result<(), BurnError> {
+        // Fetch all of the sessions
+        let sessions: Vec<DataTransferSession> =
+            sqlx::query_as("SELECT * FROM data_transfer_sessions")
+                .fetch_all(pool)
+                .await?;
+
+        // Fetch all of the sessions and group by the payer
+        let mut payer_totals = HashMap::<PublicKeyBinary, PayerTotals>::new();
+        for session in sessions.into_iter() {
+            payer_totals
+                .entry(session.payer.clone())
+                .or_default()
+                .push_sess(session);
+        }
+
+        for (
+            payer,
+            PayerTotals {
+                total_bytes,
+                sessions,
+            },
+        ) in payer_totals.into_iter()
+        {
+            // Fetch the sub dao epoch info:
+            let epoch = self.provider.get_epoch_info().await?.epoch;
+            let (sub_dao_epoch_info, _) = Pubkey::find_program_address(
+                &[
+                    "sub_dao_epoch_info".as_bytes(),
+                    self.program_cache.sub_dao.as_ref(),
+                    &epoch.to_le_bytes(),
+                ],
+                &helium_sub_daos::ID,
+            );
+
+            let amount = bytes_to_dc(total_bytes);
+
+            // Burn the DC for the payer
+            let ddc_key = crate::pdas::delegated_data_credits(&self.program_cache.sub_dao, &payer);
+            let account_data = self.provider.get_account_data(&ddc_key).await?;
+            let mut account_data = account_data.as_ref();
+            let escrow_account =
+                DelegatedDataCreditsV0::try_deserialize(&mut account_data)?.escrow_account;
+
+            tracing::info!("Burning {} DC from {}", amount, payer);
+
+            let instructions = {
+                let request = RequestBuilder::from(
+                    data_credits::id(),
+                    &self.cluster,
+                    std::rc::Rc::new(Keypair::from_bytes(&self.keypair).unwrap()),
+                    Some(CommitmentConfig::confirmed()),
+                    RequestNamespace::Global,
+                );
+
+                let accounts = accounts::BurnDelegatedDataCreditsV0 {
+                    sub_dao_epoch_info,
+                    dao: self.program_cache.dao,
+                    sub_dao: self.program_cache.sub_dao,
+                    account_payer: self.program_cache.account_payer,
+                    data_credits: self.program_cache.data_credits,
+                    delegated_data_credits: crate::pdas::delegated_data_credits(
+                        &self.program_cache.sub_dao,
+                        &payer,
+                    ),
```

Comment:
can you reuse the `ddc_key` you already computed on line 133?

Reply:
ah yes, I definitely can